Core/Src/leds/rgb_led_handler.c

```diff
@@ -40,31 +48,29 @@ void rgb_led(void* argument) {
         ECU_ENABLE_BREAKPOINT_DEBUG();
 
         // espera RTD ser setado ou timeout estourar
-        switch (
-            osMessageQueueGet(q_rgb_led_messageHandle, &message, NULL, RGB_BLINK_DELAY)) {
+        switch (osMessageQueueGet(q_rgb_led_messageHandle, &message, NULL, RGB_TIMEOUT)) {
 
             // caso timeout estore vai piscar o led, indicando que ta fora do RTD
-            case osErrorTimeout:
-                write_rgb_color(get_rgb_color(message.color));
-                blink_rgb(RGB_BLINK_DELAY);
-                break;
+            case osErrorTimeout: write_pattern(message, RGB_BLINK_200_DELAY); break;
 
             default:
                 switch (message.control) {
                     case FIXED:
-                        for (;;) {
-                            write_rgb_color(get_rgb_color(message.color));
-                            osMessageQueueGet(q_rgb_led_messageHandle, &message, NULL,
-                                              osWaitForever);
-                            if (message.control == BLINK200) {
-                                break;
+                        if (message.sizeOfPattern > 1) {
+                            for (;;) {
+                                write_pattern(message, RGB_BLINK_1000_DELAY);
+                                if (osMessageQueueGetCount(q_rgb_led_messageHandle) > 0) {
+                                    break;
+                                }
                             }
+                            break;
                         }
+                        write_rgb_color(
+                            get_rgb_color(message.pattern[ONE_COLOR_PATTERN_POS]));
+                        messageWait(q_rgb_led_messageHandle);
```

Comment:
```suggestion
                        write_rgb_color(
                            get_rgb_color(message.pattern[ONE_COLOR_PATTERN_POS]));
                        while (is_zero_messages_available(q_rgb_led_messageHandle)) {
                            osDelay(RGB_BLINK_200_DELAY);
                        }
```
Esse delay serve pra colocar a task em idle, sem ela a task rodaria o tempo todo (o que não seria problema se a prioridade dela for mínima, mas melhor colocar a task em idle).